BootloaderCommonPkg/Library/UefiVariableLib/UefiVariableLib.c

```diff
@@ -889,7 +892,7 @@ FindVariableEx (
       //
       // Record Variable in VariableIndex HOB
       //
-      if ((IndexTable != NULL) && !StopRecord) {
+      if ((IndexTable != NULL) && !StopRecord && (IndexTable->Length < UEFI_VARIABLE_INDEX_TABLE_VOLUME)) {
         Offset = (UINTN) Variable - (UINTN) LastVariable;
         if ((Offset > 0x0FFFF) || (IndexTable->Length == sizeof (IndexTable->Index) / sizeof (IndexTable->Index[0]))) {

```

Comment:
This condition may resolve the issue.
`
if ((Offset > 0x0FFFF) || (IndexTable->Length == sizeof (IndexTable->Index) / sizeof (IndexTable->Index[0]))) {
`
->
`
if ((Offset > 0x0FFFF) || (IndexTable->Length >= sizeof (IndexTable->Index) / sizeof (IndexTable->Index[0]))) {
`

Reply:
@myguitar Thanks, the new EDK2 driver already made this change, will import.